File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {

    int temp=prices[0];
    int diff=0;
    int profit=0;
    for(int i=1;i<pricesSize;i++){
        if(prices[i]-temp<0){
            temp=prices[i];
        }
        else{
            diff=prices[i]-temp;
            if(profit>diff){
                profit=profit;
            }
            else{
                profit=diff;
            }
        }
    }
    
    return profit;
    
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l=0;
        int maxprofit=0;
        int profit=0;
        for(int r=1;r<prices.length;r++){
            profit=prices[r]-prices[l];
            if(prices[r]-prices[l]<0){
                l=r;
            }
            maxprofit=Math.max(profit,maxprofit);

        }
        return maxprofit;
    }
}